TST: GH18498 - Split test_pytables into sub-modules

mikebarkas opened the pull request:
- [x] closes #18498 
- [x] tests moved / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Moved test classes for pytables into sub-modules.
Moved hdf test data files needed into sub-module.

I did not change any tests, just broke up the classes into separate modules.

Commit:
TST: GH18498: Move existing tests out of test_pytables module

  This commit moves the existing test classes out of the pytables
  test module into sub-modules.

Commit:
TST: GH18498: Removed io/test_pytables module

  These tests now live in the pytables sub-module.

Commit:
TST: GH:18498 Move legacy hdf data files into pytables sub-module

pep8speaks commented:
Hello @mikebarkas! Thanks for submitting the PR.

 - In the file [`pandas/tests/io/pytables/common.py`](https://github.com/pandas-dev/pandas/blob/65419cea814cef0b6b1cc0e35442be7ed23df383/pandas/tests/io/pytables/common.py), following are the PEP8 issues :

> [Line 28:9](https://github.com/pandas-dev/pandas/blob/65419cea814cef0b6b1cc0e35442be7ed23df383/pandas/tests/io/pytables/common.py#L28): [E722](https://duckduckgo.com/?q=pep8%20E722) do not use bare except'
> [Line 36:5](https://github.com/pandas-dev/pandas/blob/65419cea814cef0b6b1cc0e35442be7ed23df383/pandas/tests/io/pytables/common.py#L36): [E722](https://duckduckgo.com/?q=pep8%20E722) do not use bare except'
> [Line 94:5](https://github.com/pandas-dev/pandas/blob/65419cea814cef0b6b1cc0e35442be7ed23df383/pandas/tests/io/pytables/common.py#L94): [E722](https://duckduckgo.com/?q=pep8%20E722) do not use bare except'

mikebarkas commented:
PEP8 issue for `common.py` 
`do not use bare except` This is in existing code.
This issue is for moving the tests and not changing test.

jreback commented on pandas/tests/io/pytables/common.py:
```diff
@@ -0,0 +1,116 @@
+import pandas.util.testing as tm
+
+
+tables = pytest.importorskip('tables')
```
if things are not used here, remove them

jreback commented on pandas/tests/io/pytables/common.py:
```diff
@@ -0,0 +1,116 @@
+# contextmanager to ensure the file cleanup
+
+
+def safe_remove(path):
```
would like to tun these into fixtures, also this removes the need for the Base class entirely. ok with doing this in another pass though

jreback commented on pandas/tests/io/pytables/test_hdf_complex_values.py:
```diff
@@ -0,0 +1,163 @@
+import pytest
+from warnings import catch_warnings
+
```
you can just name this test_complex.py

jreback commented on pandas/tests/io/pytables/test_hdf_store.py:
```diff
@@ -11,133 +9,34 @@
-    def setup_method(self, method):
-        self.path = 'tmp.__%s__.h5' % tm.rands(10)
-
-    def teardown_method(self, method):
```
the idea is to split this out into many submodules, e.g. 

test_store (construct & instrospect a store)
test_append
test_multiindex
test_remove
test_where
test_select

basically just select similar tests (by name to start is ok) and group them into sub-modules

jreback commented on pandas/tests/io/pytables/test_timezones.py:
```diff
@@ -0,0 +1,290 @@
+
+    def test_append_with_timezones_dateutil(self):
+
+        from datetime import timedelta
```
move any imports to the top

Commit:
TST: GH:18498 Move test methods into sub-modules

  Removed test classes, created test functions
  Grouped tests into sub-modules
  Cleaned up imports, comments, and spaces

jreback commented:
@mikebarkas looks great!

oddly there is 1 less tests here (174 and 1 xfail) vs 175 and 1 xfail on master.

happy to have parameterization of any tests as well.

mikebarkas commented:
My test build on my fork is failing because  `ImportError: No module named tables`
Looking into it.

I will look for the missing test.

mikebarkas commented:
This is my missing test
```
def test_longpanel(self):
    pass
```

I will add it back into `test_panel`

jreback commented:
oh no need to add that back then

Commit:
TST: GH:18498 Conditionally test with tables import

jreback commented on pandas/tests/io/pytables/test_common.py:
```diff
@@ -0,0 +1,1131 @@
+from pandas import (Series, DataFrame, date_range, Index, Timestamp,
+                    DatetimeIndex, compat, concat, isna)
+tables = pytest.importorskip('tables')
+from pandas.tests.io.pytables.common import (ensure_clean_store, safe_close,
```
you can just do a

```
from .common import .....
```

(and do this generally)

jreback commented on pandas/tests/io/pytables/test_common.py:
```diff
@@ -0,0 +1,1131 @@
+
+import pandas.util.testing as tm
+import pandas.util._test_decorators as td
+from pandas.util.testing import (assert_panel_equal,
```
prefer not to import these and just reference with ``tm.``

jreback replied:
also do this generally across all files

jreback commented on pandas/tests/io/pytables/test_date.py:
```diff
@@ -0,0 +1,134 @@
+
+import numpy as np
+import pandas as pd
+from pandas import Series, DataFrame, date_range, timedelta_range, bdate_range
```
can you rename to test_datetime

jreback commented on pandas/tests/io/pytables/test_panel.py:
```diff
@@ -0,0 +1,33 @@
+    with catch_warnings(record=True):
+        wp = tm.makePanel()
+        _check_roundtrip(wp, assert_panel_equal)
+
```
there are may more panel tests, can you move here

jreback commented on pandas/tests/io/pytables/test_path.py:
```diff
@@ -0,0 +1,163 @@
+
+        store.close()
+        assert 'CLOSED' in store.info()
+        assert not store.is_open
```
these need to go with any other open/close tests, which I think you have in test_common.py now. call this file test_open_close (move this one and other tests)

jreback commented on pandas/tests/io/pytables/test_read.py:
```diff
@@ -0,0 +1,270 @@
+import pytest
+from warnings import catch_warnings
+
+import numpy as np
```
there are many more selection / query type tests that are in test_common. move here

jreback commented on pandas/tests/io/pytables/test_read.py:
```diff
@@ -0,0 +1,270 @@
+import pytest
```
rename this to test_query

jreback commented on pandas/tests/io/pytables/test_select.py:
```diff
@@ -0,0 +1,746 @@
+import numpy as np
+from pandas import (Series, DataFrame, date_range, Index, Timestamp,
+                    concat, MultiIndex, bdate_range, Panel)
+from pandas.tests.io.pytables.common import (ensure_clean_store,
```
consolidate with test_read (and call test_query)

jreback commented on pandas/tests/io/pytables/test_start_stop.py:
```diff
@@ -0,0 +1,85 @@
+from pandas.tests.io.pytables.common import ensure_clean_store
+import pandas.util.testing as tm
+
+
```
move these to test_select, we don't need a separate file for this

jreback commented on pandas/tests/io/pytables/test_store.py:
```diff
@@ -0,0 +1,74 @@
+    index = MultiIndex(levels=[['foo', 'bar', 'baz', 'qux'],
+                               ['one', 'two', 'three']],
+                       labels=[[0, 0, 0, 1, 1, 2, 2, 3, 3, 3],
+                               [0, 1, 2, 0, 1, 1, 2, 0, 1, 2]],
```
these need to be put in the right place, not a separate file

jreback reviewed (changes requested):
a bit too much splitting. idea is to put common files together. tests might be named inconsitenly, but still to group them together, we should end up with only about 5 ish modules here overall.

Commit:
TST: GH18498 Requested pull request changes

  Consolidate functions to reduce amount of module files.

alsobay commented:
Hi Mike, are you still working on this? Would be glad to help out or pick it up if you're too busy.

jreback commented:
closing as stale, happy to take an updated PR of this
